src/arithmetic/algebraic_expression.c

```diff
@@ -123,6 +104,41 @@ AlgebraicExpression **_AlgebraicExpression_Intermidate_Expressions(AlgebraicExpr
     return expressions;
 }
 
+/* Return index of matrix operand within algebraic expression
+ * which has the least number of none zero values. */
+int _AlgebraicExpression_MinMatPos(const AlgebraicExpression *ae) {
+    int minMatPos = 0;  // Position of matrix within ae operands.
+    GrB_Index nvals;    // number of NNZ.
+    GrB_Index minNvals; // minimum number of NNZ.
+    
+    GrB_Matrix M = ae->operands[0].operand;
+    GrB_Matrix_nvals(&minNvals, M);
+
+    // Search for minumum.
+    for(int i = 1; i < ae->operand_count; i++) {
+        M = ae->operands[i].operand;
+        GrB_Matrix_nvals(&nvals, M);
+        if(nvals < minNvals) {
+            minNvals = nvals;
+            minMatPos = i;
+        }
+    }
+    
+    return minMatPos;
+}
+
+inline void _AlgebraicExpression_Execute_MUL(GrB_Matrix C, GrB_Matrix A, GrB_Matrix B, GrB_Descriptor desc) {
```

Comment:
This function should be `static inline` - bare `inline` functions cause problems in GCC at O0.